main.js

```diff
@@ -44,127 +44,134 @@ it is available then var.The variable is available inside a block and it's child
+        }
+
+        super(Object.assign(adapterOptions, options || {}));
+    }
+
+    _unload(callback) {
+        try {
+            adapter.log.info('cleaned everything up...');
+            callback();
+        } catch (e) {
+            callback();
+        }
+    }
+
+    _objectChange(id, obj) {
+        // Warning, obj can be null if it was deleted
+        adapter.log.info('objectChange ' + id + ' ' + JSON.stringify(obj));
+    }
+
+    _stateChange(id, state) {
+        // Warning, state can be null if it was deleted
+        adapter.log.info('stateChange ' + id + ' ' + JSON.stringify(state));
+
+        // you can use the ack flag to detect if it is status (true) or command (false)
+        if (state && !state.ack) {
+            adapter.log.info('ack is not set!');
+        }
+    }
+
+    _message(obj) {
+        if (typeof obj === 'object' && obj.message) {
+            if (obj.command === 'send') {
+                // e.g. send email or pushover or whatever
+                console.log('send command');
+
+                // Send response in callback if required
+                if (obj.callback) adapter.sendTo(obj.from, obj.command, 'Message received', obj.callback);
             }
-        },
-        // is called if a subscribed object changes
-        objectChange: function (id, obj) {
-            // Warning, obj can be null if it was deleted
-            adapter.log.info('objectChange ' + id + ' ' + JSON.stringify(obj));
-        },
-        // is called if a subscribed state changes
-        stateChange: function (id, state) {
-            // Warning, state can be null if it was deleted
-            adapter.log.info('stateChange ' + id + ' ' + JSON.stringify(state));
-        
-            // you can use the ack flag to detect if it is status (true) or command (false)
-            if (state && !state.ack) {
-                adapter.log.info('ack is not set!');
-            }
-        },
-        // Some message was sent to adapter instance over message box. Used by email, pushover, text2speech, ...
-        message: function (obj) {
-            if (typeof obj === 'object' && obj.message) {
-                if (obj.command === 'send') {
-                    // e.g. send email or pushover or whatever
-                    console.log('send command');
-        
-                    // Send response in callback if required
-                    if (obj.callback) adapter.sendTo(obj.from, obj.command, 'Message received', obj.callback);
-                }
-            }
-        },
-        // is called when databases are connected and adapter received configuration.
-        // start here!
-        ready: main()
-    });
-    // you have to call the adapter function and pass a options object
-    // adapter will be restarted automatically every time as the configuration changed, e.g system.adapter.template.0
-	adapter = new utils.Adapter(options);
-
-	return adapter;
+        }
+    }
+
+    _main() {
```

Comment:
let's call it `ready`, just like the event

Reply:
will do